PictureViewMaster/PictureMasterViewController.swift

```diff
@@ -8,37 +8,84 @@
 
 import UIKit
 
-class PictureMasterViewController: UIViewController {
-
+class PictureMasterViewController: UIViewController , UIGestureRecognizerDelegate {
+    enum IncomeDirection {
+        case In
```

Comment:
May be better to use a nested enum. One for `.In` or `.Out` and inside the latter a variable holding another enum for the `.Up`, `.Down`, `.Right` and `.Left` cases.